src/admin.rs

```diff
@@ -815,4 +811,29 @@ mod tests {
             resp_str.contains(r#"waypoint":{"destination":"defaultnw/127.0.0.10","port":15008}"#)
         );
     }
+
+    pub fn test_store(
```

Comment:
The naming of this function suggests it may be a test in and of itself at first glance, consider a name which clarifies that it isn't a test

I think you moved it so it's only used in a local scope now as well so it can probably be private
```suggestion
   fn new_mock_store(
```